src/components/services/DeepLinkingManager.tsx

```diff
@@ -74,15 +80,35 @@ export function DeepLinkingManager(props: Props) {
         }
       }
 
+      /** Handler for push messages received while app is in the foreground. */
+      const handleForegroundPushMessage = (message: FirebaseMessagingTypes.RemoteMessage) => {
+        if (message.notification?.body == null) {
+          console.warn('FirebaseMessagingTypes.RemoteMessage (foreground push message) has no body')
+          return
+        }
```

Comment:
Note for later: In the `handleBackgroundPushMessage`, we use `parsePushMessage` to extract a link. We should do the same thing for foreground push messages. That way, the promo can include a link to a blog article or whatever. If the user taps the message, we should launch the link.

We can add this feature later, since right now `parsePushMessage` is extremely limited in what it can do. Once we beef up `parsePushMessage`, we can fix the dropdown to handle the link.